sycl/include/sycl/multi_ptr.hpp

```diff
@@ -27,7 +71,578 @@ template <typename dataT, int dimensions> class local_accessor;
 /// qualified pointers to allow interoperability between plain C++ and OpenCL C.
 ///
 /// \ingroup sycl_api
-template <typename ElementType, access::address_space Space> class multi_ptr {
+// TODO: Default value for DecorateAddress is for backwards compatiblity. It
+//       should be removed.
```

Comment:
Why do we need it here if we have a `multi_ptr` specialization which only takes two template arguments?

Reply:
They take all three arguments, but the legacy specialization is picked if the last one is `legacy`. Having it default here will evaluate and pick legacy over the main template definition if no additional arguments are specified.